packages/react-components/src/hooks/useAlarms/hookHelpers/useAlarmAssets.ts

```diff
@@ -0,0 +1,70 @@
+import { IoTSiteWiseClient } from '@aws-sdk/client-iotsitewise';
+import { AlarmData, AlarmRequest } from '../types';
+import { useDescribeAssetModels, useDescribeAssets } from '../../../queries';
+import {
+  buildFromAssetModelResponse,
+  buildFromAssetResponse,
+} from '../utils/alarmDataUtils';
+import { getStatusForQuery } from '../utils/queryUtils';
+import { isAssetModelRequest, isAssetRequest } from './predicates';
+
+export interface UseAlarmAssetsOptions {
+  iotSiteWiseClient?: IoTSiteWiseClient;
+  requests?: AlarmRequest[];
+}
+
+/**
+ * useAlarmAssets is a hook used to describe the asset or assetModel
+ * in an AlarmRequest and find the associated SiteWise alarms
+ *
+ * @param iotSiteWiseClient is an AWS SDK IoT SiteWise client
+ * @param requests is a list of AlarmRequests
+ * @returns a list of AlarmData for all alarms in all requests
+ */
+export function useAlarmAssets({
+  iotSiteWiseClient,
+  requests = [],
+}: UseAlarmAssetsOptions): AlarmData[] {
+  // Fetch an asset model for request with an assetModelId
+  const assetModelRequests = requests.filter(isAssetModelRequest) ?? [];
```

Comment:
I think this doesn't need to be null coalesced because requests is defaulted to a list, and filter always returns a list.

Reply:
Yes will update